plugin/src/test/java/hudson/plugins/swarm/SwarmClientIntegrationTest.java

```diff
@@ -388,6 +380,88 @@ public void missingMasterOption() throws Exception {
                         .anyMatch(line -> line.contains("Option \"-master\" is required")));
     }
 
+    @Test
+    public void workDirEnabledByDefaultWithFsRootAsDefaultPath() throws Exception {
+        final File fsRootPath = temporaryRemotingFolder.newFolder("fsrootdir");
+        TestUtils.createSwarmClient(j, processDestroyer, temporaryFolder,
+                "-fsroot", fsRootPath.getAbsolutePath());
+
+        assertTrue("Workdir exists", new File(fsRootPath, "remoting").isDirectory());
```

Comment:
In this test and the other new tests, can we strengthen the assertions to cover all expected state?

- `fsRoot` exists
- `workDir` exists (if different from `fsRoot`) or doesn't exist (if we are testing the negative case for `-failIfWorkDirIsMissing`)
- `internalDir` exists (if using working directories) or doesn't exist (if we are testing `-disableWorkDir`)
- `logs` exists (if using working directories) or doesn't exist (if we are testing `-disableWorkDir`)
- `jarCache` exists (if using working directories or `-jar-cache`) or doesn't exist (if we are testing `-disableWorkDir`)

I think the existing tests are covering most of the relevant assertions, but it would be better if we could assert all expected state in each test. Ideally it might even be possible to generalize all these assertions into a common method that is called by each test.

Reply:
Sounds good to me.

Reply:
PR is updated accordingly.